### ISSUE TITLE
add a hints-only style

### ISSUE THREAD

**Issue by nilsbecker** [tridactyl/tridactyl#2996]
Tridactyl version: 1.20.2
Firefox version: Mozilla Firefox 82.0.3
Operating system: mac

*   Brief description of the problem:

tridactyl applies heavy visual highlighting of clickable elements following vimperator but unlike vim-vixen and vimium-ff . there should be a style provided in the distribution that allows disabling this easily.

*   Steps to reproduce:

hit `f` 

*   Config (in a new tab, run `:viewconfig`, copy the url and paste it somewhere like gist.github.com):

i tried fixing this in `userContent.css` but this did not work satisfactorily, as it was not possible to fully disable element highlighting; also, after enabling `tridactylrc`, apparently userContent.css is not sourced anymore at all.

tridactylrc
```vim
" vim: filetype=vim

" delete all settings except what is here
sanitize tridactyllocal tridactylsync

" manage screen space
guiset_quiet hoverlink right

" key bindings for usability
bind d scrollpage +0.5
bind u scrollpage -0.5
bind x tabclose
bind X undo
unbind D
bind <C-j> tabmove -1
bind <C-k> tabmove +1
bind J tabprev
bind K tabnext
bind <C-i> mode ignore
bind / fillcmdline find
bind ? fillcmdline find -?
bind n findnext +1
bind N findnext -1
bind ,<Space> nohlsearch
set smoothscroll true

" from example rc files
bind gd tabdetach
bind gD composite tabduplicate; tabdetach
set newtab about:blank

" search and translate
set searchurls.translate https://translate.google.com/#view=home&op=translate&sl=%s1&tl=%s2&text=%s3
set searchurls.gs https:scholar.google.com/scholar?q=
```
usercontent.css
```css
span.TridactylHint {
	position: absolute !important;
	font-family: monospace !important;
	font-size: 14px !important;
	font-weight: bold !important;
	text-transform: lowercase !important;
	color: black !important;
	background-color: #ffd76e !important;
	border-color: rgba(0, 0, 0, 0.1) !important;
	border-width: 1px !important;
	border-radius: 2px !important;
	min-width: .75em !important;
	border-style: solid !important;
	padding: 0 1pt !important;
	text-align: center !important;
	z-index: 2147483647 !important;
}

.TridactylHintElem { 
	background-color: inherit !important;  
	border-color: rgba(0.5, 0.5, 0.5, 0.05) !important; 
} 

.TridactylHintActive { 
	background-color: rgba(1, 1, 0, 0.1) !important;
	color: black !important; 
	border-color: rgba(0, 0, 0, 0.1) !important; 
}

/** { font-family: serif ! important; }*/

```

generally speaking, i think relying on `userContent.css` is problematic as it's not synchronized between different devices, needs to be enabled explicitly by a config option with `legacy` in its name, and generally seems on the way out. a custom theme would be a  lot simpler to handle for users.

as a bonus, there could be one theme without any element highlighting as described above and another with just the element frames highlighted, which would be an intermediate solution

**Comment by bovine3dom** [comment]
I'm not sure we need a built-in theme for this when it's _fairly_ straightforward to add it as a custom theme:

`~.config/tridactyl/themes/nohl.css`:

```
:root.TridactylThemeNohl {
    --tridactyl-hint-active-bg: rgba(0,0,0,0);
    --tridactyl-hint-active-outline: 0px solid rgba(0,0,0,0);
    --tridactyl-hint-outline: 0px solid rgba(0,0,0,0);
    --tridactyl-hint-bg: 0px solid rgba(0,0,0,0);
}
```

In Tridactyl (provided `:native` is installed):
`:colours nohl`

---

I'm happy to be persuaded otherwise, though. It wouldn't be difficult to add it to Tridactyl for everyone : )

**Comment by bovine3dom** [comment]
> also, after enabling tridactylrc, apparently userContent.css is not sourced anymore at all

is very surprising to me - we shouldn't have that power. : )

**Comment by nilsbecker** [comment]
thanks for the instructions, that seems useful. i'll try it out. so i could keep the whole ~/.config/tridactyl folder synchronized via some external sync tool and get the theme on all my computers.

how easy is it to actually make this theme for a new user? i certainly would not have known how to write that css. are the relevant `--tridacty...` keys documented somewhere? if yes it might be enough.

i think it would still be good to have this as a native theme. in that way users who do not want / can't install the native client would still get a solution that auto-syncs across devices via ff sync.

**Comment by bovine3dom** [comment]
> so i could keep the whole ~/.config/tridactyl folder synchronized via some external sync tool and get the theme on all my computers.

Once it has been read, you don't need it so long as you don't run `:colours` again. You can upload it with `:firefoxsyncpush` and then download it to other machines with `:firefoxsyncpull`. But an external sync tool would also work : )

It is not easy to write a theme. I don't think we have documented it well anywhere (yet), which is why I suggested `userContent.css` first.

Let me know how you get on with the theme. If it works OK for you I'll add it to Tridactyl for everyone.

**Comment by nilsbecker** [comment]
hi, i did some testing. it almost works. i get the following undesired behavior: when hitting `f`, it's true that the hinted elements are not highlighted in any color anymore and only the letters are shown. however, they also sometimes (always?) lose their own color. for instance, on this github page, when i hit `f`, the green comment button background disappears, making the comment button invisible. (except for a thin outline}

**Comment by bovine3dom** [comment]
That isn't surprising in hindsight. I'll add making a theme that works to the to-do list : )

**Comment by itaranto** [comment]
> I'm not sure we need a built-in theme for this when it's _fairly_ straightforward to add it as a custom theme:
> 
> `~.config/tridactyl/themes/nohl.css`:
> 
> ```
> :root.TridactylThemeNohl {
>     --tridactyl-hint-active-bg: rgba(0,0,0,0);
>     --tridactyl-hint-active-outline: 0px solid rgba(0,0,0,0);
>     --tridactyl-hint-outline: 0px solid rgba(0,0,0,0);
>     --tridactyl-hint-bg: rgba(0,0,0,0);
> }
> ```
> 
> In Tridactyl (provided `:native` is installed): `:colours nohl`
> 
> I'm happy to be persuaded otherwise, though. It wouldn't be difficult to add it to Tridactyl for everyone : )

This should be the default IMHO.

**Comment by oxcl** [comment]
> > I'm not sure we need a built-in theme for this when it's _fairly_ straightforward to add it as a custom theme:
> > `~.config/tridactyl/themes/nohl.css`:
> > ```
> > :root.TridactylThemeNohl {
> >     --tridactyl-hint-active-bg: rgba(0,0,0,0);
> >     --tridactyl-hint-active-outline: 0px solid rgba(0,0,0,0);
> >     --tridactyl-hint-outline: 0px solid rgba(0,0,0,0);
> >     --tridactyl-hint-bg: rgba(0,0,0,0);
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > In Tridactyl (provided `:native` is installed): `:colours nohl`
> > I'm happy to be persuaded otherwise, though. It wouldn't be difficult to add it to Tridactyl for everyone : )
> 
> This should be the default IMHO.

agreed.

**Comment by itaranto** [comment]
> > > I'm not sure we need a built-in theme for this when it's _fairly_ straightforward to add it as a custom theme:
> > > `~.config/tridactyl/themes/nohl.css`:
> > > ```
> > > :root.TridactylThemeNohl {
> > >     --tridactyl-hint-active-bg: rgba(0,0,0,0);
> > >     --tridactyl-hint-active-outline: 0px solid rgba(0,0,0,0);
> > >     --tridactyl-hint-outline: 0px solid rgba(0,0,0,0);
> > >     --tridactyl-hint-bg: rgba(0,0,0,0);
> > > }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > In Tridactyl (provided `:native` is installed): `:colours nohl`
> > > I'm happy to be persuaded otherwise, though. It wouldn't be difficult to add it to Tridactyl for everyone : )
> > 
> > 
> > This should be the default IMHO.
> 
> agreed.

However, after taking a closer look at this style, I found that it still does _some_ highlighting.

I think this should look like the example presented here: https://github.com/tridactyl/tridactyl/issues/4106

Also, talking about the issue https://github.com/tridactyl/tridactyl/issues/4106 and https://github.com/tridactyl/tridactyl/issues/4468. I think all of these should be discussed in only one issue to avoid adding more noise.

**Comment by bovine3dom** [comment]
Fixed in #5097 

```
set hintstyles.bg none
set hintstyles.fg none
set hintstyles.outline none
```

in 1.24.2+ (unreleased at time of writing
